### PR TITLE
Get Samza to work with Java 11 (still works with Java 8)

### PR DESCRIPTION
**Pull request by james-deee**
This has 2 specific purposes. 

- Upgrade the Yarn Version to a Java 11 compatible version
- Make everything compile when building with Java 11.

I have tested that both Java 8 and Java 11 compile and all tests pass from the above.

### CODE REVIEW

File: samza-shell/src/main/bash/run-class.sh [hunk 1]
```diff
@@ -153,7 +153,7 @@ fi
 [[ $JAVA_OPTS != *-Xmx* ]] && JAVA_OPTS="$JAVA_OPTS -Xmx768M"
 
 # Check if the GC related flags are specified. If not - add the respective flags to JVM_OPTS.
-[[ $JAVA_OPTS != *PrintGCDateStamps* && $JAVA_OPTS != *-Xloggc* ]] && JAVA_OPTS="$JAVA_OPTS -XX:+PrintGCDateStamps -Xloggc:$SAMZA_LOG_DIR/gc.log"
+[[ $JAVA_OPTS != *-Xloggc* ]] && JAVA_OPTS="$JAVA_OPTS -Xloggc:$SAMZA_LOG_DIR/gc.log"
```
Comment (james-deee) [inline samza-shell/src/main/bash/run-class.sh:156]: PrintGCDateStamps is removed from Java11, so just removing it from here.

File: samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala [hunk 2]
```diff
@@ -366,16 +366,14 @@ class KafkaConfig(config: Config) extends ScalaMapConfig(config) {
     val properties = new Properties()
 
     if (isStreamMode) {
-      properties.putAll(ImmutableMap.of(
-        "cleanup.policy", "compact",
-        "segment.bytes", String.valueOf(segmentBytes),
-        "max.message.bytes", String.valueOf(maxMessageBytes)))
+      properties.put("cleanup.policy", "compact")
```
Comment (james-deee) [inline samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala:369]: The putAll method has an ambiguous issue: https://github.com/scala/bug/issues/10418

So just moving to single line puts.

**Comment by malcolm-pro** [comment]
I don't see a place to comment in the commit, but Java 11 also has removed the -d64 option:
https://github.com/james-deee/samza/blob/d6118a41bd3b4dec56776ab81a9420165a3381b7/samza-shell/src/main/bash/run-class.sh#L111

**Comment by james-deee** [comment]
Thanks. I think that's ok here though right? The test will fail on Java11,
and so will not add that flag appropriately. So I think that this check is
correct:

Java 8:
java -version
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (Temurin)(build 25.332-b09, mixed mode)

Java 11:
java -d64 -version
Unrecognized option: -d64
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.

This check seems right to me. Java8 will add the flag, but Java 11 will not
add it.

-Jamie

**Comment by james-deee** [comment]
Sorry, i meant

Java 8:
java -d64 -version
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (Temurin)(build 25.332-b09, mixed mode)

**Comment by malcolm-pro** [comment]
That seems right actually -- I thought that this might be causing some issues with my setup, but now that I'm looking at it more closely, there seems to be something else going on.

**Pull request closed without merging**